packages/grafana-ui/src/options/builder/legend.tsx

```diff
@@ -11,21 +11,33 @@ export function addLegendOptions<T extends OptionsWithLegend>(
   builder
     .addRadio({
       path: 'legend.displayMode',
-      name: 'Legend mode',
+      name: 'Mode',
       category: ['Legend'],
       description: '',
       defaultValue: LegendDisplayMode.List,
       settings: {
         options: [
           { value: LegendDisplayMode.List, label: 'List' },
           { value: LegendDisplayMode.Table, label: 'Table' },
-          { value: LegendDisplayMode.Hidden, label: 'Hidden' },
+        ],
+      },
+    })
+    .addRadio({
+      path: 'legend.showLegend',
+      name: 'Visibility',
+      category: ['Legend'],
+      description: '',
+      defaultValue: LegendVisibility.Visible,
+      settings: {
+        options: [
+          { value: LegendVisibility.Visible, label: 'Visible' },
+          { value: LegendVisibility.Hidden, label: 'Hidden' },
```

Comment:
I suggest using `addBooleanSwitch` instead and placing it above the Legend Mode option.